### ISSUE TITLE
An Interesting Issue: AMRFinder Misidentifies My Custom-Designed New NDM Variant

### ISSUE THREAD

**Issue by YuLanSky** [ncbi/amr#147]
I encountered an interesting situation while using AMRFinder (v3.12.8) to search for NDM genes in a genome. I was pleasantly surprised to see that it could accurately identify over 50 NDM subtypes, such as NDM-33, NDM-68, etc., even though these subtypes might differ by just a single amino acid.

This made me curious about what would happen if there was a novel variant not currently in the database, such as NDM-99. How would AMRFinder classify it?

To explore this, I used NDM-1 as a template and randomly modified a few sites before analyzing the sequence with AMRFinder (v3.12.8). Surprisingly, it was identified as NDM-1 with the following results:

Target length: 270
Reference sequence length: 270
% Coverage of reference sequence: 100%
% Identity to reference sequence: 100%
In summary, a modified NDM-1 sequence was still classified as NDM-1, with all scoring metrics showing 100%, which is not an accurate reflection of the actual situation. This suggests that even if a new NDM variant emerges, it might be incorrectly classified as NDM-1.

Is there a specific parameter that should be adjusted for this scenario? Thank you!

**Comment by YuLanSky** [comment]
[NDM-999.txt](https://github.com/user-attachments/files/16708325/NDM-999.txt)

**Comment by vbrover** [comment]
AMRFinderPlus identifies coding genes at protein level.
NDM-999 is blaNDM-1 at protein level.
All modifications in NDM-999 are synonymous.

**Comment by YuLanSky** [comment]
> AMRFinderPlus identifies coding genes at protein level. NDM-999 is blaNDM-1 at protein level. All modifications in NDM-999 are synonymous.

Thank you very much for your reply!

I continued to modify the sequence of NDM-999 and generated NDM-888. This time, there were some changes:

Target length: 270
Reference sequence length: 270
% Coverage of reference sequence: 100%
% Identity to reference sequence: 98.52%
However, the output still names it as NDM-1. Considering that even a single amino acid mutation would classify it as a new NDM variant, why is it still being annotated as NDM-1?

I assume there might be a rationale behind this annotation method that I'm not aware of due to my limited knowledge. Nonetheless, I would like to ask if there might be a better way to annotate it. If my question is out of place, I sincerely apologize for any offense.
[NDM-888.txt](https://github.com/user-attachments/files/16711792/NDM-888.txt)

**Comment by evolarjun** [comment]
In my hands AMRFinderPlus correctly identifies the sequence in [NDM-888.txt](https://github.com/user-attachments/files/16711792/NDM-888.txt) as an unspecified blaNDM allele. See the `Gene symbol` and `Sequence name` fields below. Possibly you are confusing the entry in the `Name of closest sequence` field which identifies the [closest reference in the database](https://www.ncbi.nlm.nih.gov/pathogens/refgene/#blaNDM-1) to the query sequence. 

```
$ amrfinder -n NDM-888.txt | d2f
Running: amrfinder -n NDM-888.txt
Software directory: '/netmnt/vast01/pd/usr/packages/AMRFinderPlus_v3.12.8/'
Software version: 3.12.8
Database directory: '/netmnt/vast01/pd/usr/packages/AMRFinderPlusData/2024-07-22.1'
Database version: 2024-07-22.1
AMRFinder translated nucleotide search
  - include -O ORGANISM, --organism ORGANISM option to add mutation searches and suppress common proteins
Running blastx
Making report
AMRFinder took 1 seconds to complete
Protein identifier Contig id Start Stop Strand Gene symbol Sequence name                                 Scope Element type Element subtype Class       Subclass   Method Target length Reference sequence length % Coverage of reference sequence % Identity to reference sequence Alignment length Accession of closest sequence Name of closest sequence                 HMM id HMM description
NA                 NDM-999     101  910      + blaNDM      NDM family subclass B1 metallo-beta-lactamase core  AMR          AMR             BETA-LACTAM CARBAPENEM BLASTX           270                       270                           100.00                            98.52              270 WP_004201164.1                subclass B1 metallo-beta-lactamase NDM-1 NA     NA
```

Note that NCBI Pathogen Detection is now the allele assigning authority for most beta-lactamase families and a few others notably including blaNDM so AMRFinderPlus should return the correct allele symbol for all blaNDM alleles unless they have just  been assigned since our last database release. See https://www.ncbi.nlm.nih.gov/pathogens/submit-beta-lactamase/ for a complete list of the families that NCBI Pathogen Detection are the allele assigning authority for. 

If, however, when you run AMRFinderPlus it doesn't return what it does when I run it on [NDM-888.txt](https://github.com/user-attachments/files/16711792/NDM-888.txt) (as above) there may be a bug, so please reopen and let us know.
